packages/headless/src/features/recent-queries/recent-queries-slice.ts

```diff
@@ -21,7 +21,7 @@ export const recentQueriesReducer = createReducer(
         state.queries = [];
       })
       .addCase(executeSearch.fulfilled, (state, action) => {
-        const query = action.payload.queryExecuted;
+        const query = action.payload.queryExecuted.trim();
         if (!query.length || state.queries.includes(query)) {
           return;
```

Comment:
Unrelated to the current story, but if the query is included in the list, but is not first, should it be moved to first place instead of a no-op?

Reply:
yes totally